ext/node/polyfills/worker_threads.ts

```diff
@@ -192,6 +192,12 @@ class NodeWorker extends EventEmitter {
       name = "[worker eval]";
     }
     this.#name = name;
+
+    const maybeWorkerData = options?.workerData;
+    const serializedWorkerData = maybeWorkerData
+      ? core.serialize(maybeWorkerData)
+      : undefined;
```

Comment:
This is wrong - we should use `serializeJsMessageData` here; and most likely include `environmentData` with `options.transferList` too.